sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp

```diff
@@ -114,10 +114,44 @@ namespace Azure { namespace Security { namespace Attestation {
    */
 
   class AttestationClient final {
-    // Allow client factory to access private methods in the AttestationClient object.
-    friend class AttestationClientFactory;
 
   public:
+    /** @brief Construct a new Attestation Client object
+     *
+     * @details Constructs a new attestation client. Follows the
+     * factory pattern in [C++ Core Guidelines
+     * C.50](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c50-use-a-factory-function-if-you-need-virtual-behavior-during-initialization)
+     *
+     * @param endpoint The URL address where the client will send the requests to.
+     * @param credential The authentication method to use (required for TPM attestation). If the
+     * credential parameter is not supplied, the connection will be unauthenticated.
+     * @param options The options to customize the client behavior.
+     * @return The newly created client.
+     */
+    static AttestationClient Create(
```

Comment:
Does it mean we don't care about instances listing the static methods for Intellisense any more?

like
AttestationClient client;
auto staticCallFromInstance = client.Create(...)

Reply:
Storage defined the static factory pattern for C++ with the `CreateFromConnectionString` API, which is a static factory method on the client.

The Attestation SDK follows the existing patterns defined for the C++ SDK. 

I'm not a huge fan, but Jeff, Anton and I (with Ahson contributing earlier) discussed it last week and decided that while it's unpleasant, the existing pattern wins.